source/mbed_trace.c

```diff
@@ -331,7 +425,8 @@ void mbed_vtracef(uint8_t dlevel, const char *grp, const char *fmt, va_list ap)
         mbed_trace_reset_tmp();
         goto end;
     }
-    if ((m_trace.trace_config & TRACE_MASK_LEVEL) &  dlevel) {
+    uint8_t config = mbed_trace_get_group_config(grp);
+    if ((config & TRACE_MASK_LEVEL) & dlevel) {
```

Comment:
This whole feature should perhaps be flagged so it wouldn't affect RAM and ROM footprint unless explicitly enabled by the user.

Reply:
It could be gated although the impact on RAM and RAM in not used is fairly minimal.

It adds 8bytes of RAM (on a 32bit target) in **total** and a few in ROM for the loop in `mbed_trace_get_group_config` function. 
Note this would not consume much runtime as in such case it would only add 2 or 3 extra instructions to fetch and compare `m_trace.group_configs_length` that'd always be 0.
A smart compiler may even be able to detech that the length is always 0 and completely discard the loop.

`mbed_trace_group_level_set` and `mbed_trace_group_level_unset` would simply not be linked in so won't cost any thing.